core/src/main/java/org/mule/runtime/core/internal/message/DefaultEventBuilder.java

```diff
@@ -481,29 +533,18 @@ private void setMessage(Message message) {
     }
 
     @Override
-    public Set<String> getVariableNames() {
-      return variables.keySet();
+    public Map<String, TypedValue<?>> getProperties() {
+      return unmodifiableMap(properties);
     }
 
     @Override
-    public <T> TypedValue<T> getVariable(String key) {
-      TypedValue<T> typedValue = variables.get(key);
-
-      if (typedValue == null) {
-        throw new NoSuchElementException("The flow variable '" + key + "' does not exist.");
-      } else {
-        return typedValue;
-      }
+    public Map<String, TypedValue<?>> getVariables() {
+      return unmodifiableMap(variables);
     }
 
     @Override
-    public Map<String, TypedValue<Object>> getProperties() {
-      return properties;
-    }
-
-    @Override
-    public Map<String, TypedValue<Object>> getParameters() {
-      return parameters;
+    public Map<String, TypedValue<?>> getParameters() {
```

Comment:
Close https://www.mulesoft.org/jira/browse/MULE-12998 (
Event API is inconsistent and implementation is not immutable after addition of properties and parameters.) when this is merged.